notion-server/src/ethereum/mod.rs

```diff
@@ -152,45 +169,84 @@ impl Simulation {
     */
 
     pub fn get_shard_block(&self, args: args::GetShardBlock) -> Result<args::ShardBlock> {
-        if let Some(shard_chain) = self.shard_chains.get(args.shard_chain_index as usize) {
-            if let Some(shard_block) = shard_chain
-                .shard_blocks
-                .get(args.shard_block_index as usize)
-            {
-                Ok(args::ShardBlock::from(shard_block))
-            } else {
-                Err(Error::OutOfBounds {
-                    message: format!(
-                        "the shard chain at index '{}' does not contain a block at index '{}'",
-                        args.shard_chain_index, args.shard_block_index
-                    ),
-                })
-            }
-        } else {
-            Err(Error::OutOfBounds {
-                message: format!(
-                    "no shard chain exists at index '{}'",
-                    args.shard_chain_index
-                ),
-            })
-        }
+        let shard_chain = self
+            .shard_chains
+            .get(args.shard_chain_index as usize)
+            .context(OutOfBounds {
+                what: WhatBound::ShardChain,
+                index: args.shard_chain_index as usize,
+            })?;
+
+        let shard_block = shard_chain
+            .shard_blocks
+            .get(args.shard_block_index as usize)
+            .context(OutOfBounds {
+                index: args.shard_block_index as usize,
+                what: WhatBound::ShardBlock(args.shard_chain_index),
+            })?;
+
+        Ok(args::ShardBlock::from(shard_block))
     }
 }
 
-pub mod args {
+mod b64_vec {
```

Comment:
I think we should call this `base64_vec`.  `b` is IMO a bit confusing, and at first glance could refer to binary, or bytes, etc.